x-pack/plugins/rule_registry/server/alert_data_client/tests/update.test.ts

```diff
@@ -0,0 +1,288 @@
+import { loggingSystemMock } from '../../../../../../src/core/server/mocks';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { elasticsearchClientMock } from 'src/core/server/elasticsearch/client/mocks';
+import { ruleDataPluginServiceMock } from '../../rule_data_plugin_service/rule_data_plugin_service.mock';
+import { RuleDataPluginServiceConstructorOptions } from '../../rule_data_plugin_service';
+import { alertingAuthorizationMock } from '../../../../alerting/server/authorization/alerting_authorization.mock';
+import { AuditLogger } from '../../../../security/server';
+
+const ruleDataServiceMock = ruleDataPluginServiceMock.create(
+  {} as RuleDataPluginServiceConstructorOptions
+);
+const alertingAuthMock = alertingAuthorizationMock.create();
+const esClientMock = elasticsearchClientMock.createElasticsearchClient();
+const auditLogger = {
+  log: jest.fn(),
+} as jest.Mocked<AuditLogger>;
+
+const alertsClientParams: jest.Mocked<ConstructorOptions> = {
+  logger: loggingSystemMock.create().get(),
+  ruleDataService: {
+    ...ruleDataServiceMock,
+    getFullAssetName: (_?: string | undefined) => '.alerts-observability-apm',
+  },
+  authorization: alertingAuthMock,
+  esClient: esClientMock,
+  auditLogger,
+};
+
+beforeEach(() => {
+  jest.resetAllMocks();
+});
+
+describe('update()', () => {
+  test('calls ES client with given params', async () => {
+    const alertsClient = new AlertsClient(alertsClientParams);
+    esClientMock.get.mockResolvedValueOnce(
+      elasticsearchClientMock.createApiResponse({
+        body: {
+          found: true,
+          _type: 'alert',
+          _index: '.alerts-observability-apm',
+          _id: 'NoxgpHkBqbdrfX07MqXV',
+          _source: {
+            'rule.id': 'apm.error_rate',
+            message: 'hello world 1',
+            'kibana.rac.alert.owner': 'apm',
+            'kibana.rac.alert.status': 'open',
+          },
+        },
+      })
+    );
+    esClientMock.update.mockResolvedValueOnce(
+      elasticsearchClientMock.createApiResponse({
+        body: {
+          _primary_term: 2,
+          result: 'updated',
+          _seq_no: 1,
+          _shards: {
+            failed: 0,
+            successful: 1,
+            total: 1,
+          },
+          _version: 1,
+          _index: '.alerts-observability-apm',
+          _id: 'NoxgpHkBqbdrfX07MqXV',
+          get: {
+            found: true,
+            _seq_no: 1,
+            _primary_term: 2,
+            _source: {
+              'rule.id': 'apm.error_rate',
+              message: 'hello world 1',
+              'kibana.rac.alert.owner': 'apm',
+              'kibana.rac.alert.status': 'closed',
+            },
+          },
+        },
+      })
+    );
+    const result = await alertsClient.update({
+      id: '1',
+      data: { status: 'closed' },
+      indexName: '.alerts-observability-apm',
+    });
+    expect(result).toMatchInlineSnapshot(`
```

Comment:
That's a lot of expectations in one unit test. I find it hard to track down failures when it's like this and usually try to limit to one expectation per test unless performance is a concern.